cordon-commands.go

```diff
@@ -0,0 +1,111 @@
+//
+// Copyright (c) 2015-2024 MinIO, Inc.
+//
+// This file is part of MinIO Object Storage stack
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+//
+
+package madmin
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"slices"
+)
+
+//msgp:ignore CordonNodeOpts
+//msgp:clearomitted
+//msgp:tag json
+//msgp:timezone utc
+//go:generate msgp -file $GOFILE
+
+const (
+	CordonAction    = "cordon"
+	UncordonAction  = "uncordon"
+	StateCordoned   = "cordoned"
+	StateUncordoned = "uncordoned"
+)
+
+func CordonActionValidate(action string) error {
+	if !slices.Contains([]string{CordonAction, UncordonAction}, action) {
+		return fmt.Errorf("invalid action '%s', must be either '%s' or '%s'", action, CordonAction, UncordonAction)
+	}
+	return nil
+}
+
+type CordonNodeOpts struct {
+	Action string
+	Node   string
+}
+
+type CordonNodeResult struct {
+	Node   string   `json:"node"`
+	Status string   `json:"status"`
+	Errors []string `json:"errors,omitempty"`
+}
+
+// Cordon will cordon a node, preventing it from receiving new requests and putting it in a maintenance mode.
+func (adm *AdminClient) Cordon(ctx context.Context, node string) (CordonNodeResult, error) {
```

Comment:
explain what is the input format for node such as is it `http://hostname:port` or `hostname:port` or just `hostname` ?